### ISSUE TITLE
Filter Slideshow content by geolocation

### ISSUE THREAD

**Issue by andyfrei** [mvysny/photocloud-frame-slideshow#129]
Hey guys
Any plans towards adding a geolocation filter? Like
- show me all pictures from Denmark
- show me all pictures from US
- show me all pictures taken in munich

Or even having an option to create a simple boundary in a map and show only pictures taken within or outside that boundary?

**Comment by mvysny** [comment]
Unfortunately this is much harder than it sounds. The problem is that every request to resolve a GPS coordinate to an Address object (which also includes the country) goes to Google servers and is bandwidth-limited. So I can't simply take a list of photos and rapidly resolve their addresses in hopes to filter the photos out, since that would most probably cause Google blocking my app from performing a GPS resolve.

Theoretically I could cache the resolved country for every photo, but the initial population of the database would be problematic - I would need to avoid rapid resolving yet somehow quickly show something.

In short, I'm not sure whether I can implement such a feature... unless there is a library that can convert GPS to country name quickly and offline. Since the country borders do not change that often :-D it could very well be possible. Let me check :-p

**Comment by mvysny** [comment]
I'll be damned, there is such a library: https://github.com/bencampion/reverse-country-code

That could work, but only for countries. There is no such (offline) library for cities.

**Comment by andyfrei** [comment]
Well, why not... ;Dandyonthemove
-------- Ursprüngliche Nachricht --------Von: Martin Vysny <notifications@github.com> Datum: 08.05.20  22:25  (GMT+01:00) An: mvysny/photocloud-frame-slideshow <photocloud-frame-slideshow@noreply.github.com> Cc: Andy Frei <andy.frei1981@gmail.com>, Author <author@noreply.github.com> Betreff: Re: [mvysny/photocloud-frame-slideshow] Filter Slideshow content by geolocation (#129) 
I'll be damned, there is such a library: https://github.com/bencampion/reverse-country-code
That could work, but only for countries. There is no such (offline) library for cities.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/mvysny/photocloud-frame-slideshow/issues/129#issuecomment-626004374",
"url": "https://github.com/mvysny/photocloud-frame-slideshow/issues/129#issuecomment-626004374",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
